src/main/java/com/jfrog/ide/common/persistency/ScanCache.java

```diff
@@ -0,0 +1,85 @@
+package com.jfrog.ide.common.persistency;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.jsontype.BasicPolymorphicTypeValidator;
+import com.fasterxml.jackson.databind.jsontype.PolymorphicTypeValidator;
+import com.jfrog.ide.common.log.Utils;
+import com.jfrog.ide.common.nodes.FileTreeNode;
+import org.jfrog.build.api.util.Log;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.List;
+
+import static com.jfrog.ide.common.utils.Utils.createMapper;
+
+/**
+ * Cache for Xray scan results.
+ *
+ * @author yahavi
+ */
+public class ScanCache {
```

Comment:
This class deserves a comprehensive unit tests

Reply:
You're right. They'll be added separately.